src/runtime/audience-action-flow.ts

```diff
@@ -321,6 +327,68 @@ export class AudienceActionFlow {
   }
 
   /**
+   * Populates localStorage with PPS configuration parameters based on
+   * SurveyDataTransferRequest.
+   **/
+  private async configureAnswerPpsData(
+    request: SurveyDataTransferRequest
+  ): Promise<void> {
+    const iabAudienceKey = StorageKeys.PPS_TAXONOMIES;
+    // PPS value field is optional and category may not be populated
+    // in accordance to IAB taxonomies.
+    const ppsConfigParams = request
+      .getSurveyQuestionsList()!
+      .flatMap((question) => question.getSurveyAnswersList())
+      .map((answer) => answer?.getPpsValue())
+      .filter((ppsValue) => ppsValue !== null);
+
+    if (ppsConfigParams.length === 0 || !ppsConfigParams) {
+      log(`[swg.js] No PPS values found in survey answer choices.`);
+    }
+
+    const existingIabTaxonomy = await this.storage_.get(
+      iabAudienceKey,
+      /* useLocalStorage= */ true
+    );
+    let ppsConfigParamsValues;
+    if (!existingIabTaxonomy) {
+      ppsConfigParamsValues = {values: ppsConfigParams};
+    } else {
+      const existingIabMap = this.filterDigits(existingIabTaxonomy);
+      ppsConfigParamsValues = {
+        values: existingIabMap.concat(ppsConfigParams as string[]),
+      };
+      await Promise.resolve(
+        this.storage_.remove(iabAudienceKey, /* useLocalStorage= */ true)
```

Comment:
Do we need to remove the existing item? Can we just execute `this.storage_.set` to overwrite?